# Restaurant Branch Performance EDA



In [1]:
import pandas as pd

df = pd.read_csv("Day13_Restaurant_Branch_Performance_Dataset.csv")
df.head()

,Record_ID,Date,Day,Month,Branch,Region,Store_Type,Staff_Count,Marketing_Spend,Customers,...,Revenue,Food_Cost,Operating_Cost,Profit,Avg_Delivery_Min,Customer_Rating,Temperature_C,Humidity_Percent,Weather,Promotion
0,ST0001,2026-06-06,Saturday,June,Jaipur,North,Express,7,4977.25,172,...,66766.77,27954.29,46554.90,20211.87,29.0,4.44,20.8,70.3,Clear,NaN
1,ST0002,2026-02-10,Tuesday,February,Pune,West,Premium,14,8998.30,285,...,157230.89,63816.59,96813.95,60416.94,27.9,4.56,31.3,70.7,Clear,Loyalty Offer
2,ST0003,2026-03-07,Saturday,March,Delhi,North,Premium,16,10034.97,306,...,155811.96,54874.10,89888.91,65923.05,30.2,4.36,21.4,61.5,Clear,NaN
3,ST0004,2026-05-11,Monday,May,Mumbai,West,Standard,9,5095.84,198,...,80574.05,31429.44,53531.82,27042.23,21.7,4.65,22.9,40.4,Clear,NaN
4,ST0005,2026-01-14,Wednesday,January,Kochi,South,Express,9,6681.82,148,...,56579.89,21759.51,44376.25,12203.64,26.1,4.50,29.0,44.3,Clear,Combo Deal


In [2]:

df.shape

(350, 22)

In [3]:

df.columns

Index(['Record_ID', 'Date', 'Day', 'Month', 'Branch', 'Region', 'Store_Type',
       'Staff_Count', 'Marketing_Spend', 'Customers', 'Orders', 'Average_Bill',
       'Revenue', 'Food_Cost', 'Operating_Cost', 'Profit', 'Avg_Delivery_Min',
       'Customer_Rating', 'Temperature_C', 'Humidity_Percent', 'Weather',
       'Promotion'],
      dtype='object')

In [4]:

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 350 entries, 0 to 349
Data columns (total 22 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Record_ID         350 non-null    object 
 1   Date              350 non-null    object 
 2   Day               350 non-null    object 
 3   Month             350 non-null    object 
 4   Branch            350 non-null    object 
 5   Region            350 non-null    object 
 6   Store_Type        350 non-null    object 
 7   Staff_Count       350 non-null    int64  
 8   Marketing_Spend   350 non-null    float64
 9   Customers         350 non-null    int64  
 10  Orders            350 non-null    int64  
 11  Average_Bill      350 non-null    float64
 12  Revenue           350 non-null    float64
 13  Food_Cost         350 non-null    float64
 14  Operating_Cost    350 non-null    float64
 15  Profit            350 non-null    float64
 16  Avg_Delivery_Min  350 non-null    float64
 1

In [5]:

df.isnull().sum()

,0
Record_ID,0
Date,0
Day,0
Month,0
Branch,0
Region,0
Store_Type,0
Staff_Count,0
Marketing_Spend,0
Customers,0


In [6]:

df.duplicated().sum()

np.int64(0)

There are **350 records** in the dataset. The dataset contains both numerical and categorical variables. I checked missing values and duplicate rows before starting the main analysis. The `Promotion` column contains missing values, while no duplicate rows were found.

## Summary Statistics

In [7]:

df.describe()

,Staff_Count,Marketing_Spend,Customers,Orders,Average_Bill,Revenue,Food_Cost,Operating_Cost,Profit,Avg_Delivery_Min,Customer_Rating,Temperature_C,Humidity_Percent
count,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000
mean,11.957143,5516.347743,217.377143,187.054286,388.906371,86831.799657,31807.545114,56621.444486,30210.355171,26.098857,4.469314,27.740000,61.324571
std,4.338261,2050.906967,56.722524,50.558371,82.488196,36644.453901,13896.888942,18467.842215,19508.592192,4.388125,0.198258,4.903148,13.757322
min,5.000000,725.350000,77.000000,71.000000,180.000000,20540.760000,7262.010000,20782.380000,-6924.680000,13.900000,3.760000,14.500000,25.000000
25%,9.000000,4339.727500,180.250000,155.000000,337.315000,61261.717500,22326.662500,43797.670000,16915.492500,22.800000,4.320000,24.400000,51.325000
50%,11.000000,5393.370000,213.000000,182.500000,383.485000,79766.880000,28861.325000,53723.025000,26873.870000,26.100000,4.470000,27.800000,61.650000
75%,15.000000,6748.502500,255.750000,218.750000,441.765000,102976.185000,38370.370000,63961.227500,39823.742500,29.075000,4.600000,31.000000,70.375000
max,24.000000,11784.960000,360.000000,334.000000,623.280000,194018.920000,75946.590000,111489.140000,89096.190000,38.300000,4.960000,43.500000,95.000000


In [8]:

print("Mean Revenue:", df["Revenue"].mean())
print("Median Revenue:", df["Revenue"].median())
print("Mode Revenue:", df["Revenue"].mode()[0])

print("\nMean Profit:", df["Profit"].mean())
print("Median Profit:", df["Profit"].median())
print("Mode Profit:", df["Profit"].mode()[0])

print("\nMean Customers:", df["Customers"].mean())
print("Median Customers:", df["Customers"].median())

Mean Revenue: 86831.79965714287
Median Revenue: 79766.88
Mode Revenue: 20540.76

Mean Profit: 30210.35517142857
Median Profit: 26873.870000000003
Mode Profit: -6924.68

Mean Customers: 217.37714285714284
Median Customers: 213.0


In [9]:

print("Customers skewness:", df["Customers"].skew())
print("Revenue skewness:", df["Revenue"].skew())
print("Profit skewness:", df["Profit"].skew())
print("Marketing Spend skewness:", df["Marketing_Spend"].skew())
print("Customer Rating skewness:", df["Customer_Rating"].skew())

Customers skewness: 0.23482621553141186
Revenue skewness: 0.8620269730776782
Profit skewness: 0.8097953191539569
Marketing Spend skewness: 0.1039906206075816
Customer Rating skewness: 0.06708477455685231


In [10]:

print("Customer count range:")
print(df["Customers"].min(), "to", df["Customers"].max())

print("\nRevenue range:")
print(df["Revenue"].min(), "to", df["Revenue"].max())

print("\nProfit range:")
print(df["Profit"].min(), "to", df["Profit"].max())

print("\nCustomer rating range:")
print(df["Customer_Rating"].min(), "to", df["Customer_Rating"].max())

Customer count range:
77 to 360

Revenue range:
20540.76 to 194018.92

Profit range:
-6924.68 to 89096.19

Customer rating range:
3.76 to 4.96


Skewness helps us understand the shape of a numerical distribution. A positive value indicates right skewness, while a negative value indicates left skewness. The ranges also help identify how spread out the values are.

## Categorical Analysis

In [11]:

df["Branch"].value_counts()

,count
Branch,
Srinagar,52
Jaipur,48
Kochi,48
Bengaluru,46
Mumbai,44
Hyderabad,43
Pune,37
Delhi,32


In [12]:

df["Region"].value_counts()

,count
Region,
South,137
North,132
West,81


In [13]:

df["Store_Type"].value_counts()

,count
Store_Type,
Standard,196
Express,102
Premium,52


In [14]:

df["Promotion"].value_counts(dropna=False)

,count
Promotion,
NaN,184
Weekend Offer,74
Combo Deal,59
Loyalty Offer,33


# Branch-wise Performance

In [15]:

branch_performance = df.groupby("Branch")[[
    "Customers", "Revenue", "Profit", "Marketing_Spend",
    "Staff_Count", "Avg_Delivery_Min", "Customer_Rating"
]].mean().round(2)

branch_performance

,Customers,Revenue,Profit,Marketing_Spend,Staff_Count,Avg_Delivery_Min,Customer_Rating
Branch,,,,,,,
Bengaluru,227.50,90732.12,31890.91,5413.93,12.85,26.56,4.47
Delhi,213.97,87573.87,31385.30,5762.63,11.56,25.69,4.49
Hyderabad,220.79,88804.90,31402.98,5455.15,12.28,25.52,4.47
Jaipur,223.27,86746.16,30469.70,5356.64,11.71,26.58,4.47
Kochi,209.06,85574.62,28902.04,5660.43,11.83,25.84,4.42
Mumbai,213.77,80779.25,26580.99,5580.10,10.89,26.86,4.44
Pune,214.92,88107.49,31265.57,5590.57,11.86,25.53,4.54
Srinagar,214.73,86746.47,30302.92,5413.67,12.46,25.97,4.48


In [16]:

branch_performance.sort_values("Profit", ascending=False)

,Customers,Revenue,Profit,Marketing_Spend,Staff_Count,Avg_Delivery_Min,Customer_Rating
Branch,,,,,,,
Bengaluru,227.50,90732.12,31890.91,5413.93,12.85,26.56,4.47
Hyderabad,220.79,88804.90,31402.98,5455.15,12.28,25.52,4.47
Delhi,213.97,87573.87,31385.30,5762.63,11.56,25.69,4.49
Pune,214.92,88107.49,31265.57,5590.57,11.86,25.53,4.54
Jaipur,223.27,86746.16,30469.70,5356.64,11.71,26.58,4.47
Srinagar,214.73,86746.47,30302.92,5413.67,12.46,25.97,4.48
Kochi,209.06,85574.62,28902.04,5660.43,11.83,25.84,4.42
Mumbai,213.77,80779.25,26580.99,5580.10,10.89,26.86,4.44


In [17]:

branch_performance.sort_values("Revenue", ascending=False)

,Customers,Revenue,Profit,Marketing_Spend,Staff_Count,Avg_Delivery_Min,Customer_Rating
Branch,,,,,,,
Bengaluru,227.50,90732.12,31890.91,5413.93,12.85,26.56,4.47
Hyderabad,220.79,88804.90,31402.98,5455.15,12.28,25.52,4.47
Pune,214.92,88107.49,31265.57,5590.57,11.86,25.53,4.54
Delhi,213.97,87573.87,31385.30,5762.63,11.56,25.69,4.49
Srinagar,214.73,86746.47,30302.92,5413.67,12.46,25.97,4.48
Jaipur,223.27,86746.16,30469.70,5356.64,11.71,26.58,4.47
Kochi,209.06,85574.62,28902.04,5660.43,11.83,25.84,4.42
Mumbai,213.77,80779.25,26580.99,5580.10,10.89,26.86,4.44


### Region-wise Performance

In [18]:

region_performance = df.groupby("Region")[[
    "Customers", "Revenue", "Profit", "Marketing_Spend",
    "Avg_Delivery_Min", "Customer_Rating"
]].mean().round(2)

region_performance

,Customers,Revenue,Profit,Marketing_Spend,Avg_Delivery_Min,Customer_Rating
Region,,,,,,
North,217.65,86946.94,30625.96,5477.53,26.13,4.48
South,218.93,88320.22,30690.57,5513.23,25.98,4.45
West,214.30,84126.72,28720.86,5584.88,26.25,4.49


In [19]:

region_performance[["Revenue", "Profit"]].sort_values("Revenue", ascending=False)

,Revenue,Profit
Region,,
South,88320.22,30690.57
North,86946.94,30625.96
West,84126.72,28720.86


## Store Type Analysis

In [20]:

store_performance = df.groupby("Store_Type")[[
    "Customers", "Orders", "Average_Bill", "Revenue",
    "Profit", "Marketing_Spend", "Avg_Delivery_Min",
    "Customer_Rating"
]].mean().round(2)

store_performance

,Customers,Orders,Average_Bill,Revenue,Profit,Marketing_Spend,Avg_Delivery_Min,Customer_Rating
Store_Type,,,,,,,,
Express,161.29,137.94,325.48,52327.16,12742.14,5122.66,26.17,4.46
Premium,298.73,258.10,514.67,152867.31,62890.24,8065.56,26.40,4.46
Standard,224.98,193.77,388.55,87268.67,30630.79,5044.90,25.98,4.47


In [21]:

store_performance.sort_values("Profit", ascending=False)

,Customers,Orders,Average_Bill,Revenue,Profit,Marketing_Spend,Avg_Delivery_Min,Customer_Rating
Store_Type,,,,,,,,
Premium,298.73,258.10,514.67,152867.31,62890.24,8065.56,26.40,4.46
Standard,224.98,193.77,388.55,87268.67,30630.79,5044.90,25.98,4.47
Express,161.29,137.94,325.48,52327.16,12742.14,5122.66,26.17,4.46


## Bivariate Analysis

In [22]:
# Studying the relationship between important pairs of variables

print("Customers and Revenue:")
print(df["Customers"].corr(df["Revenue"]))

print("\nMarketing Spend and Profit:")
print(df["Marketing_Spend"].corr(df["Profit"]))

print("\nStaff Count and Customers:")
print(df["Staff_Count"].corr(df["Customers"]))

print("\nDelivery Time and Customer Rating:")
print(df["Avg_Delivery_Min"].corr(df["Customer_Rating"]))

Customers and Revenue:
0.8763608566981173

Marketing Spend and Profit:
0.35026470499232804

Staff Count and Customers:
0.6894041595956171

Delivery Time and Customer Rating:
-0.4369837520470665


## Correlation Matrix

In [23]:
correlation_matrix = df[[
    "Staff_Count", "Marketing_Spend", "Customers", "Orders",
    "Average_Bill", "Revenue", "Food_Cost", "Operating_Cost",
    "Profit", "Avg_Delivery_Min", "Customer_Rating",
    "Temperature_C", "Humidity_Percent"
]].corr().round(2)

correlation_matrix

,Staff_Count,Marketing_Spend,Customers,Orders,Average_Bill,Revenue,Food_Cost,Operating_Cost,Profit,Avg_Delivery_Min,Customer_Rating,Temperature_C,Humidity_Percent
Staff_Count,1.00,0.28,0.69,0.67,0.52,0.70,0.68,0.79,0.56,-0.18,0.08,0.08,-0.02
Marketing_Spend,0.28,1.00,0.41,0.42,0.33,0.43,0.40,0.49,0.35,0.10,-0.06,0.06,-0.05
Customers,0.69,0.41,1.00,0.98,0.53,0.88,0.84,0.86,0.84,0.18,-0.05,0.00,-0.10
Orders,0.67,0.42,0.98,1.00,0.52,0.86,0.83,0.84,0.82,0.19,-0.05,0.02,-0.09
Average_Bill,0.52,0.33,0.53,0.52,1.00,0.84,0.84,0.80,0.83,-0.03,0.04,-0.05,-0.04
Revenue,0.70,0.43,0.88,0.86,0.84,1.00,0.98,0.96,0.97,0.08,0.01,-0.03,-0.09
Food_Cost,0.68,0.40,0.84,0.83,0.84,0.98,1.00,0.97,0.92,0.08,0.00,-0.03,-0.08
Operating_Cost,0.79,0.49,0.86,0.84,0.80,0.96,0.97,1.00,0.86,0.02,0.02,0.01,-0.07
Profit,0.56,0.35,0.84,0.82,0.83,0.97,0.92,0.86,1.00,0.13,-0.01,-0.06,-0.11
Avg_Delivery_Min,-0.18,0.10,0.18,0.19,-0.03,0.08,0.08,0.02,0.13,1.00,-0.44,0.00,-0.11


In [24]:

profit_correlation = correlation_matrix["Profit"].sort_values(ascending=False)
profit_correlation

,Profit
Profit,1.00
Revenue,0.97
Food_Cost,0.92
Operating_Cost,0.86
Customers,0.84
Average_Bill,0.83
Orders,0.82
Staff_Count,0.56
Marketing_Spend,0.35
Avg_Delivery_Min,0.13


The correlation value lies between **-1 and +1**. Values closer to +1 show a positive relationship, values closer to -1 show a negative relationship, and values close to 0 show a weak linear relationship. Correlation shows association, not cause and effect.

## Final Observations

1. **Branch performance:** Bengaluru has the highest average profit of **₹31,890.91**, while Mumbai has the lowest average profit of **₹26,580.99**.

2. **Regional performance:** The **South** region has the highest average revenue at **₹88,320.22**.

3. **Store type:** **Premium** stores have the highest average profit at **₹62,890.24**, which is much higher than the other store types.

4. **Customer volume:** **Premium** stores have the highest average customers at **298.73** customers, while Express stores have the lowest average.

5. **Customers and revenue:** Customers and Revenue have a strong positive correlation of **0.88**. This means branches with more customers generally tend to have higher revenue.

6. **Marketing spend and profit:** Marketing Spend and Profit have a positive correlation of **0.35**. This shows a relationship between the two variables, but correlation does not prove that marketing spend directly causes higher profit.

7. **Staff and customers:** Staff Count and Customers have a positive correlation of **0.69**, suggesting that branches with more staff generally handle higher customer volumes.

8. **Delivery time and rating:** Average Delivery Time and Customer Rating have a negative correlation of **-0.44**. This suggests that longer delivery times are generally associated with lower ratings in this dataset.

## Conclusion

The EDA shows that restaurant performance varies across branches, regions and store types. Customer volume is closely related to revenue, while profit is also related to several operational factors such as customers, costs and marketing spend. The branch-wise and store-type comparisons help identify better-performing groups, while the correlation matrix gives a quick view of relationships between numerical variables.

